src/test/miniminer_tests.cpp

```diff
@@ -0,0 +1,467 @@
+    for (const auto& target_feerate : various_normal_feerates) {
+        node::MiniMiner mini_miner(pool, all_unspent_outpoints);
+        BOOST_CHECK(mini_miner.IsReadyToCalculate());
+        auto bump_fees = mini_miner.CalculateBumpFees(target_feerate);
+        BOOST_CHECK(!mini_miner.IsReadyToCalculate());
+        BOOST_CHECK(sanity_check(all_transactions, bump_fees));
+        BOOST_CHECK_EQUAL(bump_fees.size(), all_unspent_outpoints.size());
+        // Check tx1 bumpfee: no other bumper.
+        const auto tx1_feerate = tx_feerates.find(tx1->GetHash())->second;
+        auto it1_unspent = bump_fees.find(COutPoint{tx1->GetHash(), 1});
+        BOOST_CHECK(it1_unspent != bump_fees.end());
+        if (target_feerate <= tx1_feerate) {
+            BOOST_CHECK_EQUAL(it1_unspent->second, 0);
+        } else {
+            // Difference is fee to bump tx1 from current to target feerate.
+            BOOST_CHECK_EQUAL(it1_unspent->second,
+                target_feerate.GetFee(tx_vsizes.find(tx1->GetHash())->second) - tx_modified_fees.find(tx1->GetHash())->second);
+        }
+        // Check tx3 bumpfee: assisted by tx4.
+        const auto tx3_feerate = CFeeRate(
+            tx_modified_fees.find(tx3->GetHash())->second + tx_modified_fees.find(tx4->GetHash())->second,
+            tx_vsizes.find(tx3->GetHash())->second + tx_vsizes.find(tx4->GetHash())->second);
+        auto it3_unspent = bump_fees.find(COutPoint{tx3->GetHash(), 1});
+        BOOST_CHECK(it3_unspent != bump_fees.end());
+        if (target_feerate <= tx3_feerate) {
+            // As long as target feerate is below tx4's ancestor feerate, there is no bump fee.
+            BOOST_CHECK_EQUAL(it3_unspent->second, 0);
+        } else {
+            // Difference is fee to bump tx3 from current to target feerate, without tx4.
+            BOOST_CHECK_EQUAL(it3_unspent->second,
+                target_feerate.GetFee(tx_vsizes.find(tx3->GetHash())->second) - tx_modified_fees.find(tx3->GetHash())->second);
+        }
+        // Check tx5 bumpfee: assisted by tx6. Specifically, tx6's modified fees.
+        const auto tx5_feerate = CFeeRate(
+            tx_modified_fees.find(tx5->GetHash())->second + tx_modified_fees.find(tx6->GetHash())->second,
+            tx_vsizes.find(tx5->GetHash())->second + tx_vsizes.find(tx6->GetHash())->second);
+        auto it5_unspent = bump_fees.find(COutPoint{tx5->GetHash(), 1});
+        BOOST_CHECK(it5_unspent != bump_fees.end());
+        if (target_feerate <= tx5_feerate) {
+            // As long as target feerate is below tx4's ancestor feerate, there is no bump fee.
+            BOOST_CHECK_EQUAL(it5_unspent->second, 0);
+        } else {
+            // Difference is fee to bump tx5 from current to target feerate, without tx6.
+            BOOST_CHECK_EQUAL(it5_unspent->second,
+                target_feerate.GetFee(tx_vsizes.find(tx5->GetHash())->second) - tx_modified_fees.find(tx5->GetHash())->second);
```

Comment:
This else-branch is currently dead code (as can easily be verified by putting an `assert(false)` inside); due to the huge modified fee-rate of tx6, tx5/tx6 are way above all of the tested fee-rate-targets (>1000sats/vbyte) and always get mini-mined, i.e. no fee-bump happens. See also review comment above.